### PR TITLE
Fix remap @ dev-1.0.0

### PR DESCRIPTION
**Pull request by cgalvarez**
## Description

- [x] Fix remap timeout or replace `it` by `xit` to ignore the test
- [x] Edit readme.md with how to use remap
- [x] Remove legacy code inside reporters because now files  are merged correctly mapped with real ones 
- [x] Edit changelog
- [x] Update version in `package.js` and `package.json`
- [x] meteor-coverage-app-exemple build ok : [Build Status](https://travis-ci.org/serut/meteor-coverage-app-exemple) 👋 
- [x] Remove annoying messages printed with `console.error()` by `remap-istanbul` when writing files of HTML report (printed as errors, but they are not; printed [here](https://github.com/gotwarlost/istanbul/blob/master/lib/report/html.js#L478-L487)).
- [ ] Issue with `console.log` and `Log.info`
- [ ] Improve coverage

### Steps to reproduce issue with `console.log` and `Log.info`

1. Replace [`warn: Log.info`](https://github.com/serut/meteor-coverage/blob/dev-1.0.0-fixes/server/report/report-remap.js#L69) with `warn: console.warn`.
2. Exec `meteor npm run test > output.txt`.
3. You'll see some messages like `Could not find ...` that `remap-istanbul` prints for every file included in the coverage that has no source map.

My intention was printing it as additional info, but when using `Log.info` or `console.log`, it prints absolutely nothing.

### Issue with remap

I should see the message `export coverage using the following format [ remap ]`, but I don't see it in the verbose output.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: server/report/report-service.js [hunk 1]
```diff
@@ -23,8 +23,8 @@ export default class {
       switch (type) {
       case 'remap':
         {
-          let reportRemap = new ReportRemap(res, type, options);
-          Conf.remap && reportRemap.generate();
+          let reportRemap = new ReportRemap(res, type, options, Log);
```
Comment (serut) [inline server/report/report-service.js]: useless parameter Log there

Comment (cgalvarez) [inline server/report/report-service.js]: Yeah, forgot removing after trying to fix the `Log.info` issue. Thanks!

File: server/report/report-remap.js [hunk 2]
```diff
@@ -28,13 +40,11 @@ export default class {
     process.chdir(Conf.COVERAGE_APP_FOLDER);
 
     // Create output directory if not exists
-    let remapFolder = path.join(Conf.COVERAGE_EXPORT_FOLDER, 'remap');
-    let remapPath = path.join(Conf.COVERAGE_APP_FOLDER, remapFolder);
-    ReportCommon.checkDirectory(remapPath);
+    ReportCommon.checkDirectory(this.remapPath);
 
-    let addFile = (filename) => path.join(remapFolder, filename);
+    let addFile = (filename) => path.join(this.remapFolder, filename);
```
Comment (serut) [inline server/report/report-remap.js]: maybe create a function `getFilePath` in the class `ReportRemap`? this.remapFolder would not be passed to that function in that case

File: server/context/conf.js [hunk 3]
```diff
@@ -85,17 +85,22 @@ if (IS_COVERAGE_ACTIVE) {
   }
 
   // Source maps remapping
+  let defFormats = ['html', 'cobertura', 'clover', 'json', 'json-summary', 'lcovonly', 'teamcity', 'text', 'text-summary'];
```
Comment (serut) [inline server/context/conf.js]: I would prefer to store these in the `/conf/default-coverage.json` file to remove these constants from the `Conf.js` file

File: server/context/conf.js [hunk 4]
```diff
@@ -85,17 +85,22 @@ if (IS_COVERAGE_ACTIVE) {
...
       }
     } else {
-      configuration['remap'] = null;
+      configuration.remap = null;
       Log.error('Loading default configuration: remap is not an object', configuration.remap);
     }
   }
+  // Assign a default value (generate all available formats) if not provided
+  if (!configuration.remap) {
+    configuration.remap = {format: defFormats};
```
Comment (serut) [inline server/context/conf.js]: if there is only remap.format = [], then rename the `remap` key into `remapFormat`

File: server/report/report-service.js [hunk 5]
```diff
@@ -1,13 +1,13 @@
+import Log from './../context/log';
+import Conf from './../context/conf';
```
Comment (serut) [inline server/report/report-service.js]: Conf isn't necessary

Comment (cgalvarez) [inline server/report/report-service.js]: The tests about exporting coverage fail if I remove it, so I'll leave it :wink:

**Pull request merged**